src/DotVVM.Framework/Hosting/ErrorPages/ErrorPageTemplate.cs

```diff
@@ -1,705 +1,324 @@
+        public const string InternalCssResourceName = "DotVVM.Framework.Resources.Styles.DotVVM.Internal.css";
 
-    public string ErrorDescription { get; set; }
+        public ErrorPageTemplate(
+            int errorCode,
+            string errorDescription,
+            string summary,
+            IErrorSectionFormatter[] formatters)
+        {
+            ErrorCode = errorCode;
+            ErrorDescription = errorDescription;
+            Summary = summary;
+            Formatters = formatters;
+        }
 
-    public string Summary { get; set; }
+        public int ErrorCode { get; }
+        public string ErrorDescription { get; }
+        public string Summary { get; }
+        public IErrorSectionFormatter[] Formatters { get; }
 
-    public IErrorSectionFormatter[] Formatters { get; set; }
+        public void WriteText(string str)
+        {
+            Write(WebUtility.HtmlEncode(str));
+        }
 
-    public void WriteText(string str)
-    {
-        Write(WebUtility.HtmlEncode(str));
-    }
+        public void WriteUnencoded(string str)
+        {
+            Write(str);
+        }
+        
+        public string TransformText()
+        {
+            // head
+            Write(
+$@"<!DOCTYPE html>
+<html>
+    <head>
+        <title>Server Error in Application</title>
+        <meta charset=""UTF-8"" />
+        <style type=""text/css"">
+");
+            using (var cssStream = typeof(DotvvmConfiguration).Assembly.GetManifestResourceStream(InternalCssResourceName))
+            using (var cssReader = new StreamReader(cssStream))
+            {
+                WriteLine(cssReader.ReadToEnd());
+            }
 
-    public void WriteUnencoded(string str)
-    {
-        Write(str);
-    }
+            foreach (var f in Formatters)
+            {
+                WriteLine($"#menu_radio_{f.Id}:checked ~ #container_{f.Id} {{ display: block; }}");
+                WriteLine($"#menu_radio_{f.Id}:checked ~ label[for='menu_radio_{f.Id}'] {{ background-color: #2980b9; }}");
+                f.WriteHead(this);
+            }
+            Write(
+@"
+        </style>
+    </head>
+");
 
-    private void WriteException(Exception ex)
-    {
-        using (var sr = new StringReader(ex.ToString()))
-        {
-            string line;
-            while ((line = sr.ReadLine()) != null)
+            // body
+            WriteUnencoded(
+$@"
+    <body>
+        <h1>Server Error, HTTP {ErrorCode}: {WebUtility.HtmlEncode(ErrorDescription)}</h1>
+        <p class=""summary"">{WebUtility.HtmlEncode(Summary)}</p>
+        <hr />
+        <div>
+");
+            foreach (var f in Formatters)
             {
-                this.WriteText(line);
-                this.Write("<br />");
+                var checkedStr = f == Formatters[0] ? "checked='checked'" : "";
+                WriteUnencoded(
+$@"
+            <input type=""radio"" id=""menu_radio_{f.Id}"" class=""collapse"" name=""menu"" {checkedStr} />
```

Comment:
```suggestion
            <input type=radio id='menu_radio_{f.Id}' class=collapse name=menu {checkedStr} />
```

And you can use single quotes when it's necessary (or if you hate unquoted attributes)